samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/bean/questionpool/QuestionPoolBean.java

```diff
@@ -2319,6 +2371,560 @@ public String getOwner() {
+		
+		String escapedFilename = org.sakaiproject.util.Validator.escapeUrl(fileName);
+        HttpServletRequest request = (HttpServletRequest) FacesContext.getCurrentInstance().getExternalContext().getRequest();
+		String userAgent = request.getHeader("User-Agent"); 
+		if (userAgent != null && userAgent.contains("MSIE")) { 
+            response.setHeader("Content-disposition", "attachment; filename=" + escapedFilename + extension);
+		}
+		else {
+        	response.setHeader("Content-disposition", "attachment; filename*=utf-8''" + escapedFilename + extension);
+        }
+		
+		OutputStream out = null;
+		try {
+			out = response.getOutputStream();
+			getAsWorkbook(spreadsheetData).write(out);
+			out.flush();
+		} catch (IOException e) {
+			if (log.isErrorEnabled()) log.error(e.getMessage());
+		} finally {
+			try {
+				if (out != null) out.close();
+			} catch (IOException e) {
+				if (log.isErrorEnabled()) log.error(e.getMessage());
+			}
+		}
+	}
+	
+	protected Workbook getAsWorkbookTest(List<List<Object>> spreadsheetData) {
+		Workbook wb = new HSSFWorkbook();
+		Sheet sheet = wb.createSheet();
+		Iterator<List<Object>> dataIter = spreadsheetData.iterator();
+
+		// By convention, the first list in the list contains column headers.
+		Row headerRow = sheet.createRow((short)0);
+		List<Object> headerList = dataIter.next();
+		for (short i = 0; i < headerList.size(); i++) {
+			createCell(headerRow, i, null).setCellValue(headerList.get(i).toString());
+		}
+		short rowPos = 1;
+		while (dataIter.hasNext()) {
+			List<Object> rowData = dataIter.next();
+			Row row = sheet.createRow(rowPos++);
+			for (short i = 0; i < rowData.size(); i++) {
+				Cell cell = createCell(row, i, null);
+				Object data = rowData.get(i);
+				if (data != null) {
+					if (data instanceof Double) {
+						cell.setCellValue(((Double)data).doubleValue());
+					} 
+					else {
+						cell.setCellValue(data.toString());
+					}
+				}
+			}
+		}
+		return wb;
+	}
+	
+	public Workbook getAsWorkbook(List<List<Object>> spreadsheetData) {
+        // outer list is rows, inner list is columns (cells in the row)
+	    int columns = findColumnSize(spreadsheetData);
+		Workbook wb;
+		if (columns < 255) {
+            log.info("Samigo export ("+columns+" columns): Using xsl format");
+		    wb = new HSSFWorkbook();
+		} else {
+		    // allows for greater than 255 columns - SAK-16560
+		    log.info("Samigo export ("+columns+" columns): Using xslx format");
+		    wb = new XSSFWorkbook();
+		}
+
+		CellStyle boldStyle = wb.createCellStyle();
+		Font font = wb.createFont();
+		font.setBoldweight(Font.BOLDWEIGHT_BOLD);
+		boldStyle.setFont(font);
+		CellStyle headerStyle = boldStyle;
+		
+		Sheet sheet = null;
+
+		Iterator<List<Object>> dataIter = spreadsheetData.iterator();
+		
+		short rowPos = 0;
+		while (dataIter.hasNext()) {
+			List<Object> rowData = dataIter.next();
+
+			if (rowData.get(0).toString().equals(ExportResponsesBean.NEW_SHEET_MARKER)) {
```

Comment:
Flip the operands of this conditional so it's null safe:

`if( ExportResponsesBean.NEW_SHEET_MARKER.equals( rowData.get( 0 ).toString() )`